### PR TITLE
Update ResourceType list format

### PR DESCRIPTION
**Pull request by janelletavares**
rather than a slice of strings, each type has a new definition
{
 "uuid": "abdfef-1234-asdasndas-de343",
 "name": "postgres",
 "opted_in": false,
 "has_access": true,
 "cli_only": false,
 "release_stage": "beta",
"categories": "",
"form_config": "{"label": "Human Readable"}"
}

related to https://github.com/meroxa/cli/issues/596

### CODE REVIEW

File: pkg/meroxa/resource_types.go [hunk 1]
```diff
@@ -6,27 +6,107 @@ import (
...
 
+type ResourceType struct {
+	UUID         string                   `json:"uuid"`
+	Name         string                   `json:"name"`
+	ReleaseStage ResourceTypeReleaseStage `json:"release_stage"`
+	Categories   []string                 `json:"categories"`
+	FormConfig   map[string]interface{}   `json:"form_config"`
+	OptedIn      bool                     `json:"opted_in"`
+	HasAccess    bool                     `json:"has_access"`
+	CLIOnly      bool                     `json:"cli_only"`
```
Comment (jayjayjpg) [inline pkg/meroxa/resource_types.go:105]: Does it make a difference in the struct if it is `CLIOnly` or `CliOnly`? From the way this is cased for other properties, I would have expected that the API attribute `cli_only` translates to `CliOnly`

If this key is basically only an internal reference and could have any value, then nevermind me here! :-)

**Review by jayjayjpg — APPROVED** [review]
This reads great, thank you for the super quick addition! :sparkles:

File: pkg/meroxa/resource_types.go (hunk 1 above)
Comment (janelletavares) [reply pkg/meroxa/resource_types.go:105]: good call, CliOnly would be more consistent

Comment (janelletavares) [reply pkg/meroxa/resource_types.go:105]:
```suggestion
	CliOnly      bool                     `json:"cli_only"`
```

**Review by ericcheatham — APPROVED** [review]

**Pull request merged**